### PR TITLE
MONGOID-5642 Queryable::Selector#merge! does not properly handle :$in/:$nin keys

### PR DESCRIPTION
**Pull request by JohnMaguir**
…ge $in and $nin when provided as a symbol as well as a string.

Issue was introduced with [PR-4645](https://github.com/mongodb/mongoid/pull/4645)
Behaviour before allowed $in and $nin to be provided as symbols or strings and would match correctly.
After [PR-4645](https://github.com/mongodb/mongoid/pull/4645) only $in or $nin strings would be merged.

This was breaking behaviour for our application during our upgrade from Mongoid 6.4 to 7.0 and is not mentioned anywhere as breaking in changelogs.

I have updated tests and new tests are passing with this change.

### CODE REVIEW

File: lib/mongoid/criteria/queryable/selector.rb [hunk 1]
```diff
@@ -22,9 +22,9 @@ def merge!(other)
             if value.is_a?(Hash) && self[key.to_s].is_a?(Hash)
               value = self[key.to_s].merge(value) do |_key, old_val, new_val|
                 case _key
```
Comment (johnnyshields) [inline lib/mongoid/criteria/queryable/selector.rb:24]: its better to do `_key.to_s` here. You regex would technically match `"I'm a $ninja"` as a key.

Comment (JohnMaguir) [reply lib/mongoid/criteria/queryable/selector.rb]: Thanks a million, you're right.
Done now

File: lib/mongoid/criteria/queryable/selector.rb [hunk 2]
```diff
@@ -21,10 +21,10 @@ def merge!(other)
           other.each_pair do |key, value|
             if value.is_a?(Hash) && self[key.to_s].is_a?(Hash)
               value = self[key.to_s].merge(value) do |_key, old_val, new_val|
-                case _key
-                when '$in'
+                case _key.to_s
+                when  '$in'
```
Comment (johnnyshields) [inline lib/mongoid/criteria/queryable/selector.rb:25]: remove extra whitespace here and below

Comment (JohnMaguir) [reply lib/mongoid/criteria/queryable/selector.rb]: Done now

**Comment by johnnyshields** [comment]
@comandeo this looks good to merge.

**Comment by comandeo** [comment]
https://jira.mongodb.org/browse/MONGOID-5642

**Review by jamis — APPROVED** [review]

**Comment by comandeo** [comment]
Hi @JohnMaguir, thank you very much for your contribution! I am sorry that it has taken us so long to deal with this PR.
Tests are green (evergreen does not report the status again).

**Pull request merged**